### ISSUE TITLE
Feature request: tf.data.Dataset.unordered_merge()

### ISSUE THREAD

**Issue by deasuke** [tensorflow/tensorflow#17471]
We're dealing with sequence models and we are bucketing training data by using
`tf.contrib.data.group_by_window.`
However, it's sequential by nature and too slow. 
So we want to call it in parallel giving up completeness of shuffle. The `tf.data.Dataset` API has
`shard` operation, however, it doesn't have `unordered_merge.` The `zip` operation waits all coming datasets and so isn't useful for this purpose. unordered_merge takes data from any comming dataset ready. It can be used to parallelize data flows.

Following is only for satisfying the format; This request is independent of the System Information.
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: YES
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source, HEAD of master 20180226
- **TensorFlow version (use command below)**:source, HEAD of master 20180226
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**: 0.10.0; installed by bazel-0.10.0-installer-linux-x86_64.sh
- **GCC/Compiler version (if compiling from source)**:4.8.5
- **CUDA/cuDNN version**:V9.1.85
- **GPU model and memory**:Tesla V100/16GB
- **Exact command to reproduce**:N/A

**Comment by mrry** [comment]
I'd welcome this as a contribution in `tf.contrib.data`.

**Comment by tamimaddari82** [comment]
I would like to work on this issue. @deasuke, Can you clarify a bit more about your use case and unordered_merge?

**Comment by deasuke** [comment]
Thank you!  @toshirin is working on it and achieved x8 throughput on 10 parallel dataset merge.
https://github.com/toshirin/tensorflow   (see unordered_merge branch)

Have to add I/F, unit tests and documents.

**Comment by toshirin** [comment]
I submitted a pull request.
Would you review it?

**Comment by deasuke** [comment]
@mrry 
Would you show me how to add  my confirm to the pull request by @toshirin ?
Will I got a mail or message on Github from Googlebot?


I think I already signed the CLA.
> The good news is that everyone that needs to sign a CLA (the pull request submitter and all commit authors) have done so. Everything is all good there.

However, the Googlebot reports this error.
> The bad news is that it appears that one or more commits were authored by someone other than the pull request submitter. We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that here in the pull request.

**Comment by rthadur** [comment]
As contrib folder is deprecated , this feature request is not relevant , so closing this issue. Than you
